### ISSUE TITLE
bad element attribute parsing

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#432]
Looking at it.wikipedia.org/wiki/Henry_Cavendish I noticed the following
![portale](https://user-images.githubusercontent.com/861631/56042369-d2545180-5d32-11e9-93bb-54d15de70be5.gif)
The 'Portale' pages are spread across the page in mediawiki and not in xowa

The html responsible is
For mediawiki

    <table class="noprint" style="width:100%;">
For xowa

    <table class="noprint">
xowa is missing the width instruction

Investigating further this leads to the Module Portale
Line 119 reads

    	local reply = { '<table class="noprint" style="width:100%;>'}
It would seem this should work

However looking closely the style attribute is missing the closing quote, it should read

    	local reply = { '<table class="noprint" style="width:100%;">'}
With that change, the visual difference disappears

I tried to make the change to itwiki Modulo:Portale but I do not have the necessary privileges

It got me to wondering as to how mediawiki copes with this (and what xowa does)

I discovered Mwh_atr_parser.java which seems to be a generic element attribute parser.
There is even a comment about dangling attributes (line 49)

Experimenting with enwiki sandbox, I think the behaviour of xowa is not correct

wikitext

    <table style="width:100%;> 
produces (first mediawiki, then xowa)

    <table style="width:100%;">
    <table>
wikitext

    <table style="width:100%; id=fred> 
produces

    <table style="width:100%; id=fred">
    <table id="fred">
wikitext

    <table style='width:100%; id=fred> 
produces

    <table style="width:100%; id=fred">
    <table id="fred">
Perhaps mediawiki entity attribute parsing logic has changed since 2014

**Comment by gnosygnu** [comment]
Yeah, XOWA has a different version of HtmlSanitizer. I tried to retrofit some behavior afterwards, but there are still discrepancies.

This will take a while to figure out. What bothers me is the two &lt;table> tags which emerge from the single one. This may need a change to the table parser, which won't be easy.

**Comment by desb42** [comment]
I may have confused with the way I was showing things
T|wo \<table> tags do not appear
I was trying to show how mediawiki does it and the how xowa does it

**Comment by gnosygnu** [comment]
> I may have confused with the way I was showing things

Nah, my mistake, I read too quickly.

If it's just a matter of re-interpreting the attributes, the fix will be much easier. I'm still working on #413 though, so it'll still be a while...
